client.go

```diff
@@ -439,7 +452,42 @@ func (m *Client) Decode(tx *types.Transaction, txErr error) (*DecodedTransaction
 	}
 
 	l := L.With().Str("Transaction", tx.Hash().Hex()).Logger()
-	receipt, err := m.WaitMined(context.Background(), l, m.Client, tx)
+
+	// if transaction was not mined, we will retry it with gas bumping, but only if gas bumping is enabled
+	// and if the transaction was not mined in time, other errors will be returned as is
+	var receipt *types.Receipt
+	err := retry.Do(
+		func() error {
+			var err error
+			ctx, cancel := context.WithTimeout(context.Background(), m.Cfg.Network.TxnTimeout.Duration())
+			receipt, err = m.WaitMined(ctx, l, m.Client, tx)
+			cancel()
+
+			return err
+		}, retry.OnRetry(func(i uint, retryErr error) {
+			replacementTx, replacementErr := prepareReplacementTransaction(m, tx)
+			if replacementErr != nil {
+				L.Debug().Str("Replacement error", replacementErr.Error()).Str("Current error", retryErr.Error()).Uint("Attempt", i).Msg("Failed to prepare replacement transaction. Retrying without the original one")
+				return
+			} else {
+				L.Debug().Str("Current error", retryErr.Error()).Uint("Attempt", i).Msg("Waiting for transaction to be confirmed after gas bump")
+			}
+			tx = replacementTx
+		}),
+		retry.DelayType(retry.FixedDelay),
+		// unless attempts is at least 1 retry.Do won't execute at all
+		retry.Attempts(func() uint {
+			if m.Cfg.GasBumpRetries() == 0 {
```

Comment:
Is it strictly 1 attempt still or 2 attempts, 1 initial and 1 after?

Reply:
that's the thing, if attempts is `0` no retry will ever be done